merkle/simple_map.go

```diff
@@ -0,0 +1,26 @@
+package merkle
+
+type SimpleMap struct {
+	kvz KVPairs
+}
+
+func NewSimpleMap() *SimpleMap {
+	return &SimpleMap{
+		kvz: nil,
+	}
+}
+
+func (sm *SimpleMap) Set(k string, v interface{}) {
+	sm.kvz = append(sm.kvz, KVPair{Key: k, Value: v})
+}
+
+// Merkle root hash of items sorted by key.
+// NOTE: Behavior is undefined when key is duplicate.
+func (sm *SimpleMap) Hash() []byte {
+	sm.kvz.Sort()
+	kvPairsH := []Hashable{}
+	for _, kvp := range sm.kvz {
```

Comment:
and what if the value is not `Hashable`? I am asking because in `KVPair#Hash` there is an if `if kvH, ok := kv.Value.(Hashable); ok {`

Reply:
The KVPair is a Hashable that knows how to deal with binary objects.